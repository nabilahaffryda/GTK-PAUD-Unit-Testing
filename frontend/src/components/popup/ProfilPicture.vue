<template>
  <avatar-cropper
    :cropper-options="options"
    :labels="labels"
    :trigger="trigger"
    :upload-form-name="nama"
    :upload-url="useBase64 ? '' : uploadUrl"
    :upload-handler="useBase64 ? onUpload : false"
    @uploaded="onUploaded"
    @changed="onChange"
  />
</template>

<script>
import AvatarCropper from 'vue-avatar-cropper';
export default {
  name: 'ProfilPicture',
  components: { AvatarCropper },
  props: {
    title: {
      type: String,
      default: '',
    },
    trigger: {
      type: String,
      default: '',
      required: true,
    },
    uploadUrl: {
      type: String,
      default: '',
    },
    useBase64: {
      type: Boolean,
      default: false,
    },
    outputMime: {
      type: String,
      default: 'image/jpeg',
    },
    nama: {
      type: String,
      default: 'foto',
    },
    limit: {
      type: Number,
      default: 2048,
    },
  },
  data() {
    return {
      labels: { submit: 'Simpan', cancel: 'Batal' },
      options: {
        aspectRatio: 4 / 6,
        autoCropArea: 4 / 6,
        viewMode: 1,
        movable: true,
        zoomable: true,
      },
    };
  },
  methods: {
    onUpload(cropper) {
      let imgBase64 = cropper.getCroppedCanvas().toDataURL(this.outputMime);
      const uploaddata = new FormData();
      uploaddata.append(this.nama, imgBase64);

      if (!imgBase64.match(/data:image\/(gif|jpeg|png);base64,(.*)/i)) {
        this.$error('Format file yang diunggah tidak sesuai');
        return;
      }

      this.$emit('upload', uploaddata);
    },
    onUploaded(resp) {
      this.$emit('uploaded', resp);
    },
    onChange(file) {
      if (file && file.size >= this.limit * 1000)
        this.$error(
          `Ukuran File gambar yang dipilih melebihi ketentuan yang diperbolehkan.
         Pilih file dengan ukuran kurang dari ${Math.round(this.limit / 1024)} MB`
        );
    },
  },
};
</script>

<style>
.avatar-cropper-overlay {
  z-index: 999 !important;
}
</style>
