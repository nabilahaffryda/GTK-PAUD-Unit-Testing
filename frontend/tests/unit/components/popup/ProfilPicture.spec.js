import { mount, createLocalVue } from "@vue/test-utils";
import ProfilPicture from "@/components/popup/ProfilPicture.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';
import AvatarCropper from 'vue-avatar-cropper';
import VueRouter from 'vue-router';

Vue.use(Vuetify)
const localVue = createLocalVue();
localVue.use(VueRouter)
const router = new VueRouter();

describe('ProfilPicture.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })

    function wrapperFactory({ } = {}) {
        return mount(ProfilPicture, {
            localVue,
            vuetify,
            router,
            stubs: {
                AvatarCropper
            },
            propsData: {
                trigger: 'trigger'
            },
            props: {
                useBase64: true
            },
        });
    }

    test('call onUpload method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.onUpload = jest.fn();
        wrapper.vm.onUpload();
        expect(wrapper.vm.onUpload.mock.calls.length).toBe(1);
    })
    test('call onUploaded method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.onUploaded = jest.fn();
        wrapper.vm.onUploaded();
        expect(wrapper.vm.onUploaded.mock.calls.length).toBe(1);
    })
    test('call onChange method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.onChange = jest.fn();
        wrapper.vm.onChange();
        expect(wrapper.vm.onChange.mock.calls.length).toBe(1);
    })
})
