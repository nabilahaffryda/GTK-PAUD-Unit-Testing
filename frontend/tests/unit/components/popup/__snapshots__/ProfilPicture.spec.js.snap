// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilPicture.vue render popup profil picture 1`] = `
<div class="avatar-cropper">
  <!----><input accept="image/png, image/gif, image/jpeg, image/bmp, image/x-icon" type="file" class="avatar-cropper-img-input">
</div>
`;
